pandas/core/ops.py

```diff
@@ -354,24 +356,25 @@ class _TimeOp(_Op):
     """
     Wrapper around Series datetime/time/timedelta arithmetic operations.
     Generally, you should use classmethod ``_Op.get_op`` as an entry point.
+
+    This is only reached in cases where either `self.is_datetime_lhs`
+    or `self.is_timedelta_lhs`.
     """
     fill_value = iNaT
 
     def __init__(self, left, right, name, na_op):
         super(_TimeOp, self).__init__(left, right, name, na_op)
 
-        lvalues = self._convert_to_array(left, name=name)
-        rvalues = self._convert_to_array(right, name=name, other=lvalues)
+        lvalues = self._convert_to_array(left)
+        rvalues = self._convert_to_array(right, other=lvalues)
 
         # left
-        self.is_offset_lhs = is_offsetlike(left)
         self.is_timedelta_lhs = is_timedelta64_dtype(lvalues)
         self.is_datetime64_lhs = is_datetime64_dtype(lvalues)
         self.is_datetime64tz_lhs = is_datetime64tz_dtype(lvalues)
         self.is_datetime_lhs = (self.is_datetime64_lhs or
                                 self.is_datetime64tz_lhs)
-        self.is_integer_lhs = left.dtype.kind in ['i', 'u']
-        self.is_floating_lhs = left.dtype.kind == 'f'
+        assert left.dtype.kind not in ['i', 'u', 'f'], left
```

Comment:
assert one of the is_* e.g.
``asert self.is_timedelta_lhs or self is_datetime_lhs``